src/VisualStudio/Core/Def/Implementation/PullMemberUp/MainDialog/PullMemberUpViewModel.cs

```diff
@@ -0,0 +1,69 @@
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp;
+using Microsoft.CodeAnalysis.PullMemberUp;
+using Microsoft.VisualStudio.LanguageServices.Implementation.Utilities;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.PullMemberUp.MainDialog
+{
+    internal class PullMemberUpViewModel : AbstractNotifyPropertyChanged
+    {
+        public ImmutableArray<PullUpMemberSymbolViewModel> Members { get; set; }
+
+        public ImmutableArray<BaseTypeTreeNodeViewModel> Destinations { get; set; }
+
+        private BaseTypeTreeNodeViewModel _selectedDestination;
+
+        public BaseTypeTreeNodeViewModel SelectedDestination { get => _selectedDestination; set => SetProperty(ref _selectedDestination, value, nameof(SelectedDestination)); }
+
+        public ImmutableDictionary<ISymbol, AsyncLazy<ImmutableArray<ISymbol>>> DependentsMap;
+
+        public ImmutableDictionary<ISymbol, PullUpMemberSymbolViewModel> SymbolToMemberViewMap { get; }
+
+        public CancellationTokenSource CancellationTokenSource { get; }
+
+        private bool _okButtonEnabled;
+
+        public bool OkButtonEnabled { get => _okButtonEnabled; set => SetProperty(ref _okButtonEnabled, value, nameof(OkButtonEnabled)); }
+
+        private bool? _selectAllCheckBoxState;
+
+        public bool? SelectAllCheckBoxState { get => _selectAllCheckBoxState; set => SetProperty(ref _selectAllCheckBoxState, value, nameof(SelectAllCheckBoxState)); }
+
+        private bool _threeStateEnable;
+
+        public bool ThreeStateEnable { get => _threeStateEnable; set => SetProperty(ref _threeStateEnable, value, nameof(ThreeStateEnable)); }
+
+        internal PullMemberUpViewModel(
+            ImmutableArray<BaseTypeTreeNodeViewModel> destinations,
+            ImmutableArray<PullUpMemberSymbolViewModel> members,
+            ImmutableDictionary<ISymbol, AsyncLazy<ImmutableArray<ISymbol>>> dependentsMap,
+            CancellationTokenSource cts)
+        {
+            Destinations = destinations;
+            DependentsMap = dependentsMap;
+            Members = members;
+            SymbolToMemberViewMap = members.ToImmutableDictionary(memberViewModel => memberViewModel.MemberSymbol);
+            CancellationTokenSource = cts;
+        }
+
+        internal PullMembersUpAnalysisResult CreateAnaysisResult()
+        {
+            var selectedOptionFromDialog = Members.
+                WhereAsArray(memberSymbolView => memberSymbolView.IsChecked && memberSymbolView.IsCheckable).
+                SelectAsArray(memberSymbolView =>
+                    (member: memberSymbolView.MemberSymbol,
+                    makeAbstract: memberSymbolView.MakeAbstract &&
+                    memberSymbolView.IsMakeAbstractCheckable));
+
+            var result = PullMembersUpAnalysisBuilder.BuildAnalysisResult(
+                SelectedDestination.MemberSymbol as INamedTypeSymbol,
```

Comment:
how do you know the MemberSymbol is an INamedType?

Reply:
So the root cause here is all the member view model inherit from http://source.roslyn.io/#Microsoft.VisualStudio.LanguageServices/Implementation/ExtractInterface/ExtractInterfaceDialogViewModel.cs,cbb7fb82de7cfedb  this class. (Also there is a same class for it in PickMembers, maybe one should be deleted).
So if you look at 'BaseTypeTreeNodeViewModel.cs' line 31. It passing the INameTypeSymbol into its base constructor, which cast it to ISymbol.